darwin/importer/importer.py

```diff
@@ -1,17 +1,28 @@
+from multiprocessing import cpu_count
 from pathlib import Path
+from time import perf_counter
 from typing import (
     TYPE_CHECKING,
     Any,
     Callable,
     Dict,
+    Generator,
     Iterable,
     List,
     Optional,
+    Sequence,
     Set,
     Tuple,
     Union,
 )
 
+from darwin.datatypes import AnnotationFile
+from darwin.item import DatasetItem
+
+Unknown = Any  # type: ignore
+
+from mpire import WorkerPool, tqdm
```

Comment:
Yeah, we're not concluded for this ourselves, but it was a useful system for managing unpickable jobs.  Nathan and I discussed it, and the main reason I stuck with it is because the refactor from MPIRE to MP is almost as simple as switching the imports - the interface is deliberately similar.

There are other better established projects, but this has great metrics for this type of task, and overcomes the pickling issues that MP can bring, so that was the logic.  The alternative was to use something like Dask or joblib which involve very different paradigms - so that's the idea behind using it.

We get good progress bars and error interception by comparison to joblib, but yeah - aware that MPIRE might need swapping, and in that case, we'd likely use mp itself.  To do that, we have to refactor some chunks of code that we foresee being replaced relatively soon, so avoided it for now.

As JSON files contain more and more masks in RLE, we'll see file sizes shoot up anyway, and only streaming the JSON would help us.